src/telliot_feeds/reporters/tellorflex.py

```diff
@@ -96,7 +96,7 @@ async def fetch_gas_price(self, speed: str = "safeLow") -> Optional[int]:
             logger.error(f"Error fetching matic gas prices: {e}")
             return None
 
-        if speed not in prices:
+        if speed not in prices or speed is None:
```

Comment:
Should be `if speed is None or speed not in prices`, because you can't check if `None` is a dict key I think. Maybe even separate the check for `None` as it's own condition w/ it's own error msg

Also, add a test simulating the behavior